model/task/task.go

```diff
@@ -3490,9 +3491,9 @@ func (t *Task) UpdateDependsOn(status string, newDependencyIDs []string) error {
 	_, err := UpdateAll(
 		bson.M{
 			DependsOnKey: bson.M{"$elemMatch": bson.M{
-				DependencyTaskIdKey:                         t.Id,
-				DependencyStatusKey:                         status,
-				DependencyDependOnGeneratedTasksDisabledKey: false,
+				DependencyTaskIdKey:             t.Id,
+				DependencyStatusKey:             status,
+				DependencyOmitGeneratedTasksKey: false,
```

Comment:
It will work as is because the `Dependency` struct that gets computed from the parser dependencies (which do have omitempty set) doesn't use omitempty for any fields: [link](https://github.com/evergreen-ci/evergreen/pull/6088/files#diff-4b29d204064065e99ec7762e466f51f0ac8461fcc5f8fc2eeda53d87a5044060R312).  I did this for `OmitGeneratedTasks` to be consistent with `Unattainable` and `Finished`, but I suppose it makes sense to add omitempty here as well. Now we do need $ne: true, I've updated it.